blogs/posts/2024-01-17_nearest_neighbour.qmd

```diff
@@ -0,0 +1,194 @@
+---
+title: "Nearest neighbour imputation"
+author:
+  - Jacqueline Grout
+date: "2024-01-17"
+categories: [learning]
+---
+
+Recently I have been gathering data by GP practice, from a variety of different sources. The ultimate purpose of my project is to be able to report at an ICB/sub-ICB level[^1]. The various datasets cover different timescales and consequently changes in GP practices over time have left me with mismatching datasets.
+
+[^1]: An ICB (Integrated Care Board) is a statutory NHS organisation responsible for planning health services for their local populations
+
+My approach has been to take as the basis of my project a recent GP List. Later in my project I want to perform calculations at a GP practice level based on an underlying health need and the data for this need is a CHD prevalence value from a dataset which is around 8 years old, and for which there is no update or alternative. From my recent list of 6454 practices, when I match to the need dataset, I am left with 151 practices without a value for need. If I remove these practices from the analysis then this could impact the analysis by sub-icb since often a group of practices in the same area could be subject to changes, mergers and reorganisation.
+
+Here is an example for two practices:
+
+Here's the packages and some demo objects to work with:
+
+```{r}
+#| code-fold: true
+#| warning: false
+#| message: false
+#| output: false
+
+
+# Packages
+library(tidyverse)
+library(sf)
+library(tidygeocoder)
+library(leaflet)
+library(viridisLite)
+library(gt)
+
+# Create some data with two practices with no need data 
+# and a selection of practices locally with need data
+practices <- tribble(
+  ~practice_code, ~postcode, ~has_orig_need, ~value,
+  "P1","CV1 4FS", 0, NA,
+  "P2","CV1 3GB", 1, 7.3,
+  "P3","CV11 5TW", 1, 6.9,
+  "P4","CV6 3HZ", 1, 7.1,
+  "P5","CV6 1HS", 1, 7.7,
+  "P6","CV6 5DF", 1, 8.2,
+  "P7","CV6 3FA", 1, 7.9,
+  "P8","CV1 2DL", 1, 7.5,
+  "P9","CV1 4JH", 1, 7.7,
+  "P10","CV10 0GQ", 1, 7.5,
+  "P11","CV10 0JH", 1, 7.8,
+  "P12","CV11 5QT", 0, NA,
+  "P13","CV11 6AB", 1, 7.6,
+  "P14","CV6 4DD", 1,7.9
+) 
+
+# get domain of numeric data
+(domain <- range(practices$has_orig_need))
+
+# make a colour palette
+pal <- colorNumeric(palette = viridis(2), domain = domain)
+
+```
+
+To provide a suitable estimate of need for the newer practices without values, all the practices in the dataset were geocoded[^2] using the `geocode` function from the {tidygeocoder} package.
```

Comment:
add article

```suggestion
To provide a suitable estimate of the need for the newer practices without values, all the practices in the dataset were geocoded[^2] using the `geocode` function from the {tidygeocoder} package.
```

Reply:
I have changed the two sentances into one